File: RailgunVB/Core/Containers/PlayerContainer.vb
Imports Discord

Namespace Core.Music
    
    Public Class PlayerContainer
        
        Public ReadOnly Property GuildId As ULong
        Public ReadOnly Property TextChannel As ITextChannel
        Public ReadOnly Property Player As Player

        Public Sub New(tc As ITextChannel, player As Player)
            _GuildId = tc.GuildId
            _TextChannel = tc
            _Player = player
        End Sub
        
    End Class
    
End NameSpace

File: Railgun/Core/Containers/PlayerContainer.cs
using System.Threading;
using Discord;
using Railgun.Core.Music;

namespace Railgun.Core.Containers
{
    public class PlayerContainer
    {
        public ulong GuildId { get; }
        public ITextChannel TextChannel { get; }
        public Player Player { get; }
        public IUserMessage LogEntry { get; set; }
        public SemaphoreSlim Lock { get; } = new SemaphoreSlim(1);

        public PlayerContainer(ITextChannel tc, Player player) {
            GuildId = tc.GuildId;
            TextChannel = tc;
            Player = player;
        }
    }
}